@utahdts/utah-design-system/react/components/forms/CalendarInput/CalendarInput.jsx

```diff
@@ -0,0 +1,389 @@
+  // focus on the next date; delay so that the new month view draws before it focuses
+  setTimeout(
+    () => {
+      if (newDate) {
+        const formattedDate = format(newDate, dateFormat);
+        document.getElementById(`${calendarInputId}__${formattedDate}`)?.focus();
+      }
+    },
+    0
+  );
+}
+
+/**
+ * @param {object} props
+ * @param {string} [props.className]
+ * @param {string} [props.dateFormat] use `date-fns` modifiers for formatting the date
+ * @param {string} [props.defaultValue] expects value to be in format of props.dateFormat
+ * @param {string} [props.errorMessage]
+ * @param {string} props.id when tied to a Form the `id` is also the 'dot' path to the data in the form's state: ie person.contact.address.line1
+ * @param {import('react').RefObject<HTMLDivElement>} [props.innerRef]
+ * @param {boolean} [props.isDisabled]
+ * @param {boolean} [props.isHidden] a dateInput will hide its calendar popup when not in use
+ * @param {boolean} [props.isRequired]
+ * @param {string} props.label
+ * @param {string} [props.labelClassName]
+ * @param {(newValue: string) => void} [props.onChange] e => {}; can be omitted for uncontrolled OR using form's onChange
+ * @param {boolean} [props.shouldSetFocusOnMount] if rendered in a popup, then set focus to first focusable element when first shown
+ * @param {boolean} [props.showTodayButton]
+ * @param {string | null} [props.value] expects value to be in format of props.dateFormat
+ * @param {string} [props.wrapperClassName]
+ * @returns {import('react').JSX.Element}
+ */
+export function CalendarInput({
+  className,
+  dateFormat = 'MM/dd/yyyy',
+  defaultValue,
+  errorMessage,
+  id,
+  innerRef,
+  isDisabled,
+  isHidden,
+  isRequired,
+  label,
+  labelClassName,
+  onChange,
+  shouldSetFocusOnMount,
+  showTodayButton,
+  value,
+  wrapperClassName,
+  ...rest
+}) {
+  const { addPoliteMessage } = useAriaMessaging();
+  const calendarInputId = useId();
+  const firstFocusableElementRef = useRef(/** @type {any | null} */(null));
+  const {
+    onChange: currentOnChange,
+    value: currentValue,
+  } = useFormContextInputValue({
+    id,
+    defaultValue,
+    onChange,
+    value,
+  });
+
+  // currentValueDate is the currently selected date
+  const currentValueDate = currentValue ? parse(currentValue, dateFormat, new Date()) : null;
+
+  // currentValueDateInternal is the currently focused date (not necessarily the selected/value date)
+  const [currentValueDateInternal, setCurrentValueDateInternal] = useState(/** @type {Date | null} */(null));
+
+  // if new value passed in, move to that month
+  useEffect(
+    () => {
+      if (currentValueDateInternal?.getTime() !== currentValueDate?.getTime()) {
+        setCurrentValueDateInternal((currentValueDate && isValid(currentValueDate)) ? currentValueDate : new Date());
+      }
+    },
+    [currentValueDate?.getTime()]
+  );
+
+  // focus on first element when popped open
```

Comment:
Need to point out how we solved this: We created a focusable element so that it doesn't drop on the first icon button, which looked wierd.